_tutorials/2016-07-18-python-automation-netmiko-and-napalm.md

```diff
@@ -0,0 +1,226 @@
+---
+published: false
+date: '2016-07-18 23:11 +0300'
+title: 'Python automation: netmiko and napalm'
+author: Mike Korshunov
+tags:
+  - vagrant
+  - iosxr
+  - Python
+excerpt: Brief introduction to automation with Python
+---
+{% include toc icon="table" title="IOS-XR: Ansible and Vagrant" %}
+
+
+
+## Introduction
+
+At first, let's take a look at used tools:
+
+[Netmiko](https://github.com/ktbyers/netmiko) - multi-vendor ssh tool for device configuration 
+
+[NAPALM](https://github.com/napalm-automation/napalm)- tool, which provides standardized API for different platforms. 
```

Comment:
`[NAPALM](https://github.com/napalm-automation/napalm)- python based automation tool, which provides a common API for different vendor platforms.`